codebasin/finder.py

```diff
@@ -159,11 +161,31 @@ def find(
                         f"{filename} found in definition of platform {p} "
                         + "but missing from codebase",
                     )
-            state.insert_file(e["file"])
+            filenames.add(e["file"])
+    for f in tqdm(
+        filenames,
+        desc="Parsing",
+        unit="file",
+        leave=False,
+        disable=not show_progress,
+    ):
+        state.insert_file(f)
 
     # Process each tree, by associating nodes with platforms
-    for p in configuration:
-        for e in configuration[p]:
+    for p in tqdm(
+        configuration,
+        desc="Preprocessing",
+        unit="platform",
+        leave=False,
+        disable=not show_progress,
+    ):
+        for e in tqdm(
```

Comment:
Do you need a `desc` for this bar?

Reply:
There is one (line 184).

It might not be obvious because `p` is a bad variable name -- it's a string representing the platform name.